code/modules/games/cards.dm

```diff
@@ -399,16 +829,30 @@
 			turn_hand(usr)
 		else
 			if(cardUser.get_item_by_slot(SLOT_HUD_LEFT_HAND) == src || cardUser.get_item_by_slot(SLOT_HUD_RIGHT_HAND) == src)
-				pickedcard = href_list["pick"]
-				Removecard()
+				var/datum/playingcard/picked_card = locateUID(href_list["pick"])
+				if(istype(picked_card))
+					remove_card(cardUser, picked_card)
```

Comment:
There is no check in remove_card for wheter the picked card is actually from the deck or not.